src/module.c

```diff
@@ -302,6 +306,80 @@ static void* RG_GetFlatExecutionPrivateData(ExecutionCtx* ectx){
     return NULL;
 }
 
+static void RedisGears_SaveRegistrations(RedisModuleIO *rdb, int when){
+    if(when == REDISMODULE_AUX_BEFORE_RDB){
+        return;
+    }
+    Gears_dictIterator* iter = Gears_dictGetIterator(Readerdict);
+    Gears_dictEntry *curr = NULL;
+    while((curr = Gears_dictNext(iter))){
+        const char* readerName = Gears_dictGetKey(curr);
+        MgmtDataHolder* holder = Gears_dictGetVal(curr);
+        RedisGears_ReaderCallbacks* callbacks = holder->callback;
+        if(!callbacks->rdbSave){
+            continue;
+        }
+        RedisModule_SaveStringBuffer(rdb, readerName, strlen(readerName) + 1 /* for \0 */);
+        callbacks->rdbSave(rdb);
+    }
+    Gears_dictReleaseIterator(iter);
+    RedisModule_SaveStringBuffer(rdb, "", 1); // empty str mean the end!
+}
+
+static int RedisGears_LoadRegistrations(RedisModuleIO *rdb, int encver, int when){
+    if(encver > REDISGEARS_DATATYPE_VERSION){
+        RedisModule_LogIOError(rdb, "warning", "could not load rdb created with higher RedisGears version!");
+        return REDISMODULE_ERR; // could not load rdb created with higher Gears version!
+    }
+    if(when == REDISMODULE_AUX_BEFORE_RDB){
+        Gears_dictIterator* iter = Gears_dictGetIterator(Readerdict);
+        Gears_dictEntry *curr = NULL;
+        while((curr = Gears_dictNext(iter))){
+            MgmtDataHolder* holder = Gears_dictGetVal(curr);
+            RedisGears_ReaderCallbacks* callbacks = holder->callback;
+            if(!callbacks->clear){
+                continue;
+            }
+            callbacks->clear();
+        }
+        Gears_dictReleaseIterator(iter);
+    } else {
+        // when loading keys phase finished, we load the registrations.
+        for(char* readerName = RedisModule_LoadStringBuffer(rdb, NULL) ;
```

Comment:
RedisModule_LoadStringBuffer() can return NULL.

Reply:
If it returns NULL it first call moduleRDBLoadError(io); and exit...